tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,169 @@
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute or module named '{name}'")
+
+    def __len__(self) -> int:
+        r"""
+        Returns the total number of modules in the sequence.
+
+        Returns:
+            int: The number of modules in the sequence.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 64), tp.Linear(64, 128))
+            assert len(model) == 2
+        """
+        return len(self.modules)
+
+    def __iter__(self) -> Iterator[Module]:
+        r"""
+        Returns an iterator over the modules in the sequence.
+
+        Returns:
+            Iterator[Module]: An iterator over the modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            for layer in model:
+                print(layer)
+        """
+        return iter(self.modules.values())
+
+    def __getitem__(self, idx: Union[int, str]) -> Module:
+        r"""
+        Accesses a module by index (int) or name (str).
+
+        Args:
+            idx (Union[int, str]): The index or name of the module to retrieve.
+
+        Returns:
+            Module: The module at the specified index or name.
+
+        Raises:
+            TypeError: If `idx` is not an int or str.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            print(model[1])
+        """
+        if not isinstance(idx, (int, str)):
+            raise TypeError("Index must be an int or str.")
+
+        key = str(idx) if isinstance(idx, int) else idx
+
+        if key not in self.modules:
+            raise ValueError(f"Key {key} not found in modules.")
+
+        return self.modules[key]
+
+    def named_children(self) -> Iterator[Tuple[str, "Module"]]:
+        r"""
+        Returns an iterator over all child modules in this `Sequential` container.
+        Each child module is represented by its name and the module object itself.
+
+        Returns:
+            Iterator[Tuple[str, Module]]: An iterator over tuples containing
+            the name and module of each child.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            for name, child in model.named_children():
+                print(f"{name}: {type(child).__name__}")
+
+        """
+        for name, module in self.modules.items():
+            if isinstance(module, Module):
+                yield name, module
+            elif isinstance(module, Sequential):
+                # Traverse deeper into Sequential containers within Sequential
+                for child_name, child in module.named_children():
+                    yield f"{name}.{child_name}", child
```

Comment:
`named_children` should only return the first-order children of the module. It is called recursively when required.